tcms/rpc/tests/test_bug.py

```diff
@@ -0,0 +1,29 @@
+import unittest
+
+from django.conf import settings
+from django.core.cache import cache
+
+from tcms.rpc.tests.utils import APITestCase
+
+if 'tcms.bugs.apps.AppConfig' not in settings.INSTALLED_APPS:
+    raise unittest.SkipTest('tcms.bugs is disabled')
+
+
+class TestBug(APITestCase):
+
+    def test_get_details_from_cache(self):
+        url = "http://some.url"
+        expected_result = {
+            'title': 'Bug from cache',
+            'description': 'This bug came from the Django cache'
+        }
+        cache.set(url, expected_result)
+        result = self.rpc_client.Bug.details(url)
```

Comment:
In this scenario you need to verify that the value was retrieved from cache instead of the integration returning the same expected result. One possible way is to mock `tracker_from_url` and then assert that it was not called.
grep for `assert_called_once_with` and see https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_not_called

Reply:
I don't get what you are saying. In this case we care that if there is a cache entry with the same key(url) it will be returned. we don't care whether the tracker in this case is Bugzilla/ Github/ Kiwi, etc. because we never go to the code that calls that:
https://github.com/kiwitcms/Kiwi/blob/master/tcms/rpc/api/bug.py#L36-L43

Reply:
That's right, but your test snippet as it is now can't tell us whether the result was returned directly from cache without consulting the bug tracker integration or if fetching from cache returned None and the bug tracker integration returned the same result.  Relying only on the result doesn't distinguish between the two cases. That's what a mock+assert on the mock is designed to do.